TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs

```diff
@@ -37,10 +34,44 @@ public class TrsDataSyncHelper(
     };
 
     private readonly ISubject<object[]> _syncedEntitiesSubject = new Subject<object[]>();
+    private readonly NpgsqlDataSource _trsDbDataSource;
+    private readonly IOrganizationServiceAsync2 _organizationService;
+    private readonly ReferenceDataCache _referenceDataCache;
+    private readonly IClock _clock;
+    private readonly IAuditRepository _auditRepository;
+    private readonly ILogger<TrsDataSyncHelper> _logger;
+    private bool? isFakeXrm;
+
+    public TrsDataSyncHelper(
+        NpgsqlDataSource trsDbDataSource,
+        [FromKeyedServices(TrsDataSyncService.CrmClientName)] IOrganizationServiceAsync2 organizationService,
+        ReferenceDataCache referenceDataCache,
+        IClock clock,
+        IAuditRepository auditRepository,
+        ILoggerFactory loggerFactory)
+    {
+        _trsDbDataSource = trsDbDataSource;
+        _organizationService = organizationService;
+        _referenceDataCache = referenceDataCache;
+        _clock = clock;
+        _auditRepository = auditRepository;
+        _logger = loggerFactory.CreateLogger<TrsDataSyncHelper>();
+    }
 
     public IObservable<object[]> GetSyncedEntitiesObservable() => _syncedEntitiesSubject;
 
-    private bool IsFakeXrm { get; } = organizationService.GetType().FullName == "Castle.Proxies.ObjectProxy_2";
+    private bool IsFakeXrm
```

Comment:
Does this need to be expanded?

Reply:
probably not if I make the change to be able to use a primary constructor and not use the ILoggerFactory